packages/beacon-node/src/network/processor/index.ts

```diff
@@ -249,18 +251,22 @@ export class NetworkProcessor {
       if (slotRoot) {
         // DOS protection: avoid processing messages that are too old
         const {slot, root} = slotRoot;
-        if (slot < this.chain.clock.currentSlot - EARLIEST_PERMISSIBLE_SLOT_DISTANCE) {
+        const clockSlot = this.chain.clock.currentSlot;
+        const {fork} = message.topic;
+        let earliestPermissableSlot = clockSlot - DEFAULT_EARLIEST_PERMISSIBLE_SLOT_DISTANCE;
+        if (ForkSeq[fork] >= ForkSeq.deneb && topicType === GossipType.beacon_attestation) {
+          // post deneb, the attestations could be in current or previous epoch
+          earliestPermissableSlot = computeStartSlotAtEpoch(this.chain.clock.currentEpoch - 1);
+        }
```

Comment:
Should we move this into a function somewhere and be re-used in gossip + here + state transition? In a way to signal that this logic must be consistent between those three places

Reply:
+1